libyul/AsmParser.h

```diff
@@ -91,6 +103,7 @@ class Parser: public langutil::ParserBase
 
 private:
 	Dialect const& m_dialect;
+	std::optional<langutil::SourceLocation> m_locationOverride;
```

Comment:
Maybe just call it `m_location` and add a comment explaining why it has to override the other one?

Reply:
I think this is a very exceptional situation and should warrant for a long variable name.